pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -867,13 +873,19 @@ func (a *Actuator) requestPowerOff(ctx context.Context, baremetalhost *bmh.BareM
 }
 
 //requestPowerOn removes requestPowerOffAnnotation from baremetalhost which signals BMO to power on the machine
-func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMetalHost) error {
-	if baremetalhost.Annotations == nil {
-		baremetalhost.Annotations = make(map[string]string)
+func (a *Actuator) requestPowerOn(ctx context.Context, machine *machinev1beta1.Machine, baremetalhost *bmh.BareMetalHost) error {
+	if machine.Annotations == nil {
+		machine.Annotations = make(map[string]string)
 	}
+	machine.Annotations[powerOnRequestTimestampAnnotation] = time.Now().Format(annotationTimestampFormat)
 
-	if _, powerOffRequestExists := baremetalhost.Annotations[requestPowerOffAnnotation]; !powerOffRequestExists {
-		return &machineapierrors.RequeueAfterError{RequeueAfter: time.Second * 5}
+	if err := a.client.Update(ctx, machine); err != nil {
+		log.Printf("Failed to add remediation power on timestamp annotation to %s: %s", machine.Name, err.Error())
```

Comment:
Instead of logging and then returning the error here, it would be better to wrap the error with the extra context and return it so the caller can log it. That way all of the information about a failure shows up in 1 log message at the end of reconciliation.